### PR TITLE
Added change methods to CheckboxComponent

### PR DESCRIPTION
**Pull request by saberlynx**
**Checklist**:

- [] Tests
- [] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: gsa/src/web/components/form/checkbox.js [hunk 1]
```diff
@@ -49,19 +49,47 @@ const CheckboxComponent = ({
...
-        <StyledTitle data-testid="checkbox-title" disabled={disabled}>
-          {title}
-        </StyledTitle>
-      )}
-      {children}
-    </Divider>
-  </StyledElement>
-);
+}) => {
+  const [value, setValue] = useState(undefined);
```
Comment (bjoernricks) [inline gsa/src/web/components/form/checkbox.js]: I guess it should be initialized with the value of a prop named checked (i am not sure if it must be named checked or value). E.g. `<Checkbox checked={true} ... />`

File: gsa/src/web/components/form/checkbox.js [hunk 2]
```diff
@@ -73,12 +101,6 @@ CheckboxComponent.propTypes = {
...
 };
 
-export default compose(
-  withLayout(),
-  withChangeHandler({
-    convert_func: convert_checked,
-    value_func: event => event.target.checked,
-  }),
-)(CheckboxComponent);
+export default compose(withLayout())(CheckboxComponent);
```
Comment (bjoernricks) [inline gsa/src/web/components/form/checkbox.js]: You can drop the compose function if we only have one HOC left.

```suggestion
export default withLayout(CheckboxComponent);
```

File: gsa/src/web/components/form/checkbox.js [hunk 3]
```diff
@@ -49,19 +50,49 @@ const CheckboxComponent = ({
...
-        <StyledTitle data-testid="checkbox-title" disabled={disabled}>
-          {title}
-        </StyledTitle>
-      )}
-      {children}
-    </Divider>
-  </StyledElement>
-);
+}) => {
+  const [value, setValue] = useState(undefined);
```
Comment (bjoernricks) [inline gsa/src/web/components/form/checkbox.js]: Either we need to initialize value with the value from the prop or we drop the internal state completely. Dropping the internal state would make the component controlled only but that would be fine for me.

Comment (saberlynx) [reply gsa/src/web/components/form/checkbox.js]: > Dropping the internal state would make the component controlled only but that would be fine for me.

Sounds great to me!

**Pull request merged**